src/PdfSmith.BusinessLayer/Templating/HandlebarsTemplateEngine.cs

```diff
@@ -0,0 +1,122 @@
+using System.Collections.Concurrent;
+using System.Globalization;
+using System.Text.RegularExpressions;
+using HandlebarsDotNet;
+using PdfSmith.BusinessLayer.Exceptions;
+using PdfSmith.BusinessLayer.Services;
+
+namespace PdfSmith.BusinessLayer.Templating;
+
+public partial class HandlebarsTemplateEngine(TimeZoneTimeProvider timeZoneTimeProvider) : ITemplateEngine
+{
+    private readonly IHandlebars handlebars = Handlebars.Create();
+    private static readonly ConcurrentDictionary<string, HandlebarsTemplate<object, object>> TemplateCache = new();
+    private static TimeZoneTimeProvider? timeProvider;
+
+    static HandlebarsTemplateEngine()
+    {
+        RegisterGlobalHelpers();
+    }
+
+    public async Task<string> RenderAsync(string template, object model, CultureInfo culture, CancellationToken cancellationToken = default)
+    {
+        try
+        {
+            // Set the time provider for helpers
+            timeProvider = timeZoneTimeProvider;
+
+            // Replace DateTime.Now with timezone-aware helper placeholders like other engines
+            var sanitizedTemplate = DateTimeNowRegex.Replace(template, "{{dateTimeNow}}");
+            sanitizedTemplate = DateTimeOffsetNowRegex.Replace(sanitizedTemplate, "{{dateTimeOffsetNow}}");
+
+            // Get or compile template (with caching for performance)
+            var compiledTemplate = TemplateCache.GetOrAdd(sanitizedTemplate, key =>
+                handlebars.Compile(key)
+            );
+
+            cancellationToken.ThrowIfCancellationRequested();
+
+            // Pass the model directly (Handlebars expects the root model, not wrapped)
+            var result = await Task.Run(() => compiledTemplate(model), cancellationToken);
```

Comment:
If Handlerbars doesn't provide a native async method, do not wrap the code within `Task.Run`.